frame/social-swap2/src/tests.rs

```diff
@@ -0,0 +1,262 @@
+use crate::{mock::*, Error};
+use super::*;
+use frame_support::assert_noop;
+use codec::Encode;
+
+#[test]
+fn test_mint_should_not_work() {
+	new_test_ext().execute_with(|| {
+		<Reserve0<Test>>::put(INITIAL_BALANCE);
+		assert_noop!(
+			SocialSwap2::mint(
+				Origin::signed(ACCOUNT1),
+				ACCOUNT3
+			),
+			Error::<Test>::NotEnoughLiquidity
+		);
+	});
+}
+
+#[test]
+fn test_mint_should_work() {
+	new_test_ext().execute_with(|| {
+
+		SocialSwap2::initialize(Origin::root(), FEE_TO, ADDRESS0, TREASURY, TOKEN0, TOKEN1);
+		let token_0_amount: u128  = 1_000_000_000_000_000_000;
+		let token_1_amount: u128  = 4_000_000_000_000_000_000;
+		pallet_assets::Module::<Test>::transfer(&ASSET_ID, &ACCOUNT1, &TOKEN0, token_0_amount);
+		pallet_assets::Module::<Test>::transfer(&ASSET_ID, &ACCOUNT1, &TOKEN1, token_1_amount);
+		let expected_liquidity: u128 = 2_000_000_000_000_000_000u128;
+
+		assert_eq!(
+			pallet_assets::Module::<Test>::total_supply(ASSET_ID),
+			0
+		);
+		assert_eq!(
+			SocialSwap2::mint(
+				Origin::signed(ACCOUNT1),
+				ACCOUNT2
+			),
+			Ok(())
+		);
+
+		assert_eq!(
+			pallet_assets::Module::<Test>::total_supply(ASSET_ID),
+			expected_liquidity
+		);
+
+		assert_eq!(
+			pallet_assets::Module::<Test>::balance(ASSET_ID, ACCOUNT2),
+			expected_liquidity - MINIMUM_LIQUIDITY as u128
+		);
+
+		assert_eq!(
+			pallet_assets::Module::<Test>::balance(ASSET_ID, TOKEN0),
+			token_0_amount
+		);
+
+		assert_eq!(
+			pallet_assets::Module::<Test>::balance(ASSET_ID, TOKEN1),
+			token_1_amount
+		);
+
+		assert_eq!(
+			SocialSwap2::reserve0(),
+			token_0_amount
+		);
+
+		assert_eq!(
+			SocialSwap2::reserve1(),
+			token_1_amount
+		);
+	});
+}
+
+#[test]
+fn test_burn_should_not_work() {
+	new_test_ext().execute_with(|| {
+
+		assert_ne!(SocialSwap2::mint(
+				Origin::signed(ACCOUNT1),
+				ACCOUNT3
+			),Ok(()));
+
+	});
```

Comment:
can use `assert_noop!` to check error here
```
assert_noop!(SocialSwap2::mint(
				Origin::signed(ACCOUNT1),
				ACCOUNT3
			),
                        Error::<Test>::SomeError
                 );
```

Reply:
@andor0 

Ideally it should be 

```
assert_noop!(
			SocialSwap2::mint(
				Origin::signed(ACCOUNT1),
				ACCOUNT3
			),
			Error::<Test>::InsufficientLiquidityMinted
		);
```

But its giving weird error.
		
```
		---- tests::test_burn_should_not_work stdout ----
thread 'tests::test_burn_should_not_work' panicked at 'assertion failed: `(left == right)`
  left: `[128, 219, 156, 72, 65, 174, 164, 236, 63, 108, 115, 17, 130, 98, 41, 26, 225, 203, 75, 115, 97, 222, 192, 210, 45, 21, 176, 145, 212, 197, 126, 200]`,
 right: `[239, 236, 170, 140, 85, 84, 36, 71, 249, 25, 167, 223, 40, 139, 34, 146, 239, 90, 7, 71, 111, 192, 3, 123, 43, 134, 173, 115, 110, 201, 173, 166]`', frame/social-swap2/src/tests.rs:78:5
```